### DISCUSSION TITLE
Skeleton rendering children even when `isLoaded` is false?

### DISCUSSION THREAD

**Discussion by vedantroy** [chakra-ui/chakra-ui#4006] [Q&A]
Does a `Skeleton` render its children even if `isLoaded` is `false`? I notice that I have components crashing because their `data` is undefined even though the `Skeleton` has `isLoaded={data !== undefined}`

**Comment by elliothimmelfarb** [comment]
We encountered an issue related to this while attempting to write tests for the `Skeleton` component. It appears that the `Skeleton` renders its children even when `isLoaded` is `false`, which makes testing difficult as the `Skeleton` and its children remain unchanged in both states. 

If anyone has experience testing `Skeleton` components and has any suggestions, we would greatly appreciate any advice.
